pkg/network/ebpf/c/kafka/kafka-helpers.h

```diff
@@ -0,0 +1,200 @@
+#ifndef __KAFKA_HELPERS_H
+#define __KAFKA_HELPERS_H
+
+#include "kafka-types.h"
+
+// Forward declaration
+static __always_inline bool try_parse_produce_request(kafka_transaction_t *kafka_transaction);
+static __always_inline bool try_parse_fetch_request(kafka_transaction_t *kafka_transaction);
+static __always_inline bool extract_and_set_first_topic_name(kafka_transaction_t *kafka_transaction);
+
+static __inline int32_t read_big_endian_int32(const char* buf) {
+    int32_t val;
+    bpf_probe_read_kernel(&val, 4, (void*)buf);
+    return bpf_ntohl(val);
+}
+
+static __inline int16_t read_big_endian_int16(const char* buf) {
+    int16_t val;
+    // De-Referencing buf causes misalignment verifier error for unknown reason, so using bpf_probe_read_kernel as a workaround
+    bpf_probe_read_kernel(&val, 2, (void*)buf);
+    return bpf_ntohs(val);
+}
+
+static __inline int32_t kafka_read_big_endian_int32(kafka_transaction_t *kafka_transaction) {
+    const int32_t value = read_big_endian_int32(kafka_transaction->request_fragment + kafka_transaction->current_offset_in_request_fragment);
+    kafka_transaction->current_offset_in_request_fragment += 4;
```

Comment:
I sort of like your approach of having a "stateful" parser (and storing the current offset in the map), but I think there is a good chance that some of the verifier issues you mentioned to me could be an artifact of this.

If you change your code a little bit to use "hard-coded" offsets instead (if that's applicable for kafka headers), there is a chance that the verifier will better understand it, because these will become immediate values in the BPF instructions that can be analyzed statically.

UPDATE: going through the rest of the code I noticed that we're parsing some variable length fields, so I'm not sure if my advice is really applicable